.github/workflows/linux.yaml

```diff
@@ -64,3 +64,35 @@ jobs:
 
       - name: Run tests
         run: ctest --test-dir _build
+  Alpine:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        platform: ['x86', 'x86_64', 'armhf','armv7','aarch64','ppc64le','riscv64', 's390x']

```

Comment:
We don't need the x86 ones. We also don't need arvmv7 and armhf at the same time. The Little-Endian one is particularly interresting though.

Reply:
All of these are little endian unfortunately. The author of the action doesn't allow MIPS. Guess QEMU has bugs.

Reply:
In this end since there's no Big Endian, I don't think this bring anything to the table, apart from longer PR and potentially having to pay for all the CI work at some point. We don't use assembly code (and likely never will). So it's the same compiler on all these platforms. In VLC we only see issues when we switch compilers versions or a library is trying to detect which CPU it's running on (for assembler code).

Reply:
Reduced to armhf, aarch64, and s390x since those are somewhat interesting.

In other projects, passing -Werror=cast-align has resulted in various errors, mainly relating to casting between struct sockaddr* on non x86. I doubt it would be an issue here but who knows.

I've also seen errors on s390x with both libupnp and libnpupnp. No real idea why.

On an unrelated note, why are these files using CRLF line endings?

Reply:
Oh lovely:

```
 /home/runner/work/libebml/libebml/src/EbmlCrc32.cpp:234:13: error: cast from 'const libebml::binary*' {aka 'const unsigned char*'} to 'const uint32_t*' {aka 'const unsigned int*'} increases required alignment of target type [-Werror=cast-align]
  234 |     crc ^= *reinterpret_cast<const std::uint32_t *>(input);
```

Reply:
Fixed in a questionable way.